torch/csrc/jit/register_special_ops.cpp

```diff
@@ -0,0 +1,33 @@
+#include "torch/csrc/autograd/profiler.h"
+#include "torch/csrc/jit/custom_operator.h"
+#include "torch/csrc/jit/operator.h"
+
+namespace torch {
+namespace jit {
+
+namespace {
+RegisterOperators reg({
+    Operator(
+        "aten::split(Tensor self, int[] split_sizes, int dim=0) -> Tensor[]",
```

Comment:
I am hesitant to call this split (vs split_with_sizes), but I know little about jit op registry.

Reply:
`split` is exposed on the user side for both a single int and a list param (e.g. `x.split(2)` and `x.split([2, 2])`), so I figured it would need to be the same case for the corresponding aten op. 

The motivation for this was in [`test_autograd.py`](https://github.com/pytorch/pytorch/blob/master/test/test_autograd.py#L3163) some of the tests call `split` with a list.

Reply:
Yep, but that was just to preserve BC. But from someone who knows a bit about aten but not about jit, `split_with_size` is specifically named that way in aten, to avoid confusion. The name `aten::split` seems to suggest that this should do what aten `split` does, but it actually is not.

Reply:
Ofc @apaszke and @zdevito 's opinions are more important here.

Reply:
Disregarding deprecated functionality, what is API we want exposed in Python?
Is it a `split` and a `split_with_sizes` function? If so, we should just not support the `int[]`-based `split` in script (policy: don't implement already-deprecated functionality in script). 

If Python will continue to have a `split` that takes an `int[]` then we should keep it.

Reply:
`split_with_sizes` is not in public API. In python, there is a custom `def split` that dispatches to either c++ function depending on the input args. Yes, python will continue to have split that takes an `int[]`.